src/Sarif/Comparers/ComparerHelpers.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Comparers
+{
+    internal class ComparerHelper
+    {
+        /// <summary>
+        /// Compare 2 object refences. Return value false presents need to
+        /// compare objects values to get final result.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <param name="result">
+        /// 0 if both objects are same or both are null.
+        /// -1 if the first object is null and the second object is not null.
+        /// 1 if the first object is not null and the second object is null.
+        /// </param>
+        /// <returns>Return true if can get a definite compare result, otherwise return false.</returns>
+        public static bool CompareReference(object left, object right, out int result)
```

Comment:
>CompareReference [](http://example.com/codeflow?start=27&length=16)

Rename this to 'TryCompare'.

This is internal code, so you could also return a nullable int. If it's null, no result.

Reply:
renamed. 
to be consistent with other compares method i d like to return int instead of nuallable int. what do you think?